src/test/regress/sql/intermediate_results.sql

```diff
@@ -252,4 +252,66 @@ END;
 -- results should have been deleted after transaction commit
 SELECT * FROM read_intermediate_results(ARRAY['squares_1', 'squares_2']::text[], 'binary') AS res (x int, x2 int);
 
+-- test refreshing mat views
+SET client_min_messages TO ERROR;
+SELECT run_command_on_workers($$CREATE USER some_other_user;$$);
+CREATE USER some_other_user;
+SELECT run_command_on_workers($$GRANT ALL ON DATABASE regression TO some_other_user;$$);
+GRANT ALL ON DATABASE regression TO some_other_user;
+RESET client_min_messages;
+
+\c - some_other_user
+
+CREATE SCHEMA other_schema;
+SET search_path TO other_schema;
+
+CREATE TABLE dist_table (a int, b int);
+INSERT INTO dist_table(a, b) SELECT n, n+1 FROM generate_series(1, 10) n;
+SELECT create_distributed_table('dist_table', 'a');
+
+CREATE MATERIALIZED VIEW mat_view AS
+SELECT *
+FROM (
+  SELECT * FROM dist_table
+  LIMIT 50000
+) q;
+
+CREATE MATERIALIZED VIEW mat_view_2 AS
+  SELECT count(*) FROM (SELECT * FROM dist_table LIMIT 50000) q, (SELECT * FROM dist_table LIMIT 100) r WHERE q.a > r.a;
+
+REFRESH MATERIALIZED VIEW other_schema.mat_view;
+REFRESH MATERIALIZED VIEW other_schema.mat_view_2;
+
+-- Now connect back as a different user and run REFRESH MATERIALIZED VIEW command,
+-- which in turn executes a repartition join query.
+\c - postgres
+REFRESH MATERIALIZED VIEW other_schema.mat_view;
+REFRESH MATERIALIZED VIEW other_schema.mat_view_2;
+
+\c - some_other_user
+
+-- test security definer funcs
+CREATE FUNCTION security_definer_in_files()
```

Comment:
I think I really want to have one more test. 

Create `security_definer_in_files_2()` function via the postgres user. Then, use them in the same transaction of `postgres` user. So that multiple folders are created&dropped in the same transaction.


```SQL
BEGIN;
SELECT * FROM security_definer_in_files_2(), security_definer_in_files();

-- even call twice
SELECT * FROM security_definer_in_files_2(), security_definer_in_files();
COMMIT;
```

Because we don't seem to have any test coverage for that (this would have failed on the master branch, so we didn't have either)